src/core/arbitrage/arbitrage.ts

```diff
@@ -11,12 +12,23 @@ export interface OptimalTrade {
 /**
  *
  */
-export function trySomeArb(paths: Array<Path>, botConfig: BotConfig): OptimalTrade | undefined {
+export function trySomeArb(paths: Array<Path>, botConfig: BotConfig, errorpaths: Map<string,number>): OptimalTrade | undefined {
 	const optimalTrade: OptimalTrade | undefined = getOptimalTrade(paths, botConfig.offerAssetInfo);
 
 	if (!optimalTrade) {
 		return undefined;
 	} else {
+		//Path to address String
+		let addrs: any= new Array
+		//Get Addresses from arbTrade.path.Pools and add to array
+		for (let i = 0; i < optimalTrade.path.pools.length; i++) {
+			addrs.push(optimalTrade.path.pools[i].address);        
```

Comment:
for speed reasons we shouldnt do this here, order of pools doesnt change in a path, we can create this unique identifier at initialisation @nick134-bit